ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go

```diff
@@ -114,54 +63,98 @@ func create(c *cli.Context, networkMode, serviceName, clusterName string, cfnCli
 	cfnClient.WaitUntilCreateComplete(sdsStackName)
 
 	// Return the ID of the SDS we just created
-	return getOutputIDFromStack(cfnClient, sdsStackName, CFNTemplateOutputSDSARN)
+	return getOutputIDFromStack(cfnClient, sdsStackName, cfnTemplateOutputSDSARN)
 }
 
-func getOutputIDFromStack(cfnClient cloudformation.CloudformationClient, stackName, outputKey string) (*string, error) {
-	response, err := cfnClient.DescribeStacks(stackName)
-	if err != nil {
+func createNamespace(c *cli.Context, input *utils.ServiceDiscovery, networkMode, serviceName, clusterName string, cfnClient cloudformation.CloudformationClient) (*string, error) {
+	namespaceParams := namespaceCFNParams(input)
+	if err := namespaceParams.Validate(); err != nil {
 		return nil, err
 	}
-	if len(response.Stacks) == 0 {
-		return nil, fmt.Errorf("Could not find CloudFormation stack: %s", stackName)
-	}
 
-	for _, output := range response.Stacks[0].Outputs {
-		if aws.StringValue(output.OutputKey) == outputKey {
-			return output.OutputValue, nil
-		}
+	namespaceStackName := cfnStackName(privateDNSNamespaceStackNameFormat, clusterName, serviceName)
+	if _, err := cfnClient.CreateStack(cloudformation.GetPrivateNamespaceTemplate(), namespaceStackName, false, namespaceParams); err != nil {
+		return nil, err
 	}
-	return nil, fmt.Errorf("Failed to find output %s in stack %s", outputKey, stackName)
 
-}
+	logrus.Info("Waiting for the private DNS namespace to be created...")
+	// Wait for stack creation
```

Comment:
Given the log statement before and the name of the method being called on the cfnClient after, I'd say this comment is redundant.